src/components/tests/assistant-code-block.test.tsx

```diff
@@ -263,4 +280,43 @@ describe( 'createCodeComponent', () => {
 			expect( getIpcApi().openLocalPath ).not.toHaveBeenCalled();
 		} );
 	} );
+
+	describe( 'when the "copy and open terminal" button is clicked', () => {
+		it( 'should not be visible for non-bash code blocks', () => {
+			render( <CodeBlock className="language-php" children="<?php echo 'Hello'; ?>" /> );
+
+			expect( screen.queryByText( 'Copy and open terminal' ) ).not.toBeInTheDocument();
+		} );
+
+		it( 'should be visible for bash code blocks', () => {
+			render( <CodeBlock className="language-bash" children="wp plugin list" /> );
+
+			expect( screen.getByText( 'Copy and open terminal' ) ).toBeInTheDocument();
+		} );
+
+		it( 'should be visible for sh code blocks', () => {
+			render( <CodeBlock className="language-sh" children="wp plugin list" /> );
+
+			expect( screen.getByText( 'Copy and open terminal' ) ).toBeInTheDocument();
+		} );
+
+		it( 'should copy the code content to the clipboard and open terminal', async () => {
+			const mockCopyText = jest.fn();
+			const mockOpenTerminalAtPath = jest.fn();
+			( getIpcApi as jest.Mock ).mockReturnValue( {
+				copyText: mockCopyText,
+				openTerminalAtPath: mockOpenTerminalAtPath,
+			} );
```

Comment:
Nitpick: We don't strictly need to create the mocked functions, as they will accessible from the import.

```suggestion
			( getIpcApi as jest.Mock ).mockReturnValue( {
				copyText: jest.fn(),
				openTerminalAtPath: jest.fn(),
			} );
```

Then you can use the following in the expectations statements:
```
expect( getIpcApi().copyText ).toHaveBeenCalledWith( 'wp plugin list' );
expect( getIpcApi().openTerminalAtPath ).toHaveBeenCalledWith(
	selectedSite.path,
	expect.any( Object )
);
```

Reply:
Removed here! https://github.com/Automattic/studio/pull/563/commits/cffef8dbdc8774c7c3f46d89afd5c8360b95a5d6